datafusion/core/tests/sqllogictests/test_files/aggregate.slt

```diff
@@ -36,6 +36,53 @@ STORED AS CSV
 WITH HEADER ROW
 LOCATION '../../testing/data/csv/aggregate_test_100.csv'
 
+#######
+# Error tests
+#######
+
+# https://github.com/apache/arrow-datafusion/issues/3353
+statement error Aggregations require unique expression names
+SELECT approx_distinct(c9) count_c9, approx_distinct(cast(c9 as varchar)) count_c9_str FROM aggregate_test_100
+
+# csv_query_approx_percentile_cont_with_weight()
+statement error Error during planning: The function ApproxPercentileContWithWeight does not support inputs of type Utf8.
+SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
+
+statement error  Error during planning: The weight argument for ApproxPercentileContWithWeight does not support inputs of type Utf8
+SELECT approx_percentile_cont_with_weight(c3, c1, 0.95) FROM aggregate_test_100
+
+statement error Error during planning: The percentile argument for ApproxPercentileContWithWeight must be Float64, not Utf8
+SELECT approx_percentile_cont_with_weight(c3, c2, c1) FROM aggregate_test_100
+
+
+# csv_query_approx_percentile_cont_with_weight
```

Comment:
Looks like duplicated block, except double spaces on [line 51](https://github.com/apache/arrow-datafusion/pull/5055/files#diff-d44e3605231ebbdef511cc254b4fe6d6d84af632275e98daefc155e3db0181efR51) and dot in the end of [line 48](https://github.com/apache/arrow-datafusion/pull/5055/files#diff-d44e3605231ebbdef511cc254b4fe6d6d84af632275e98daefc155e3db0181efR48)

Reply:
And it also seems each of these 3 error messages should end with dot, though sqllogictests completely  ignores actual error message - even
```
statement error just some random text here
SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
```
counts as passed test 🤔

Reply:
Good call with duplicated block. Thank you 🙏 



As for the error messages, I am not sure what you mean by "though sqllogictests completely ignores actual error message - even"

I tried changing the expected error message like this

```diff
diff --git a/datafusion/core/tests/sqllogictests/test_files/aggregate.slt b/datafusion/core/tests/sqllogictests/test_files/aggregate.slt
index 3b527dcd8..6931018da 100644
--- a/datafusion/core/tests/sqllogictests/test_files/aggregate.slt
+++ b/datafusion/core/tests/sqllogictests/test_files/aggregate.slt
@@ -56,7 +56,7 @@ SELECT approx_percentile_cont_with_weight(c3, c2, c1) FROM aggregate_test_100
 
 
 # csv_query_approx_percentile_cont_with_weight
-statement error Error during planning: The function ApproxPercentileContWithWeight does not support inputs of type Utf8.
+statement error some random text
 SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
 
 statement error Error during planning: The weight argument for ApproxPercentileContWithWeight does not support inputs of type Utf8
```

And then test failed, as I expected:
```shell
cargo test --test sqllogictests
....
STORED AS CSV
WITH HEADER ROW
LOCATION '../../testing/data/csv/aggregate_test_100.csv'"
[aggregate.slt] Running query: "SELECT approx_distinct(c9) count_c9, approx_distinct(cast(c9 as varchar)) count_c9_str FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c3, c1, 0.95) FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c3, c2, c1) FROM aggregate_test_100"
[aggregate.slt] Running query: "SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100"
Error: statement is expected to fail with error:
	some random text
but got error:
	DataFusion error: Error during planning: The function ApproxPercentileContWithWeight does not support inputs of type Utf8.
[SQL] SELECT approx_percentile_cont_with_weight(c1, c2, 0.95) FROM aggregate_test_100
at tests/sqllogictests/test_files/aggregate.slt:59

error: test failed, to rerun pass `-p datafusion --test sqllogictests`
```

🤔

Reply:
Sorry, I should have checked this more careful - it was unsaved version of file.

Now I see that it's ok if expected result is leftmost substring of SQL execution result, so it perfectly explains why missing dot in the end of string doesn't affect test results.

Reply:
Thank you for double checking @korowa